### PR TITLE
[WIP] Update for JupyterLab 2.0

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
cc @telamonian @jasongrout.  FYI, we did run into the TS 3.7 issue with accessors:

```
##[error]node_modules/@jupyterlab/application/lib/lab.d.ts(42,9): error TS1086: An accessor cannot be declared in an ambient context.
```

**Comment by telamonian** [comment]
Okay, I think I understand the TS 3.7 issue completely now.

The declaration `.d.t.s` files generated by TS 3.7 [now explicitly include `get` and `set` method declarations for accessors](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier), instead of just pretending that accessors are class props. The part that I was missing before is that [versions of Typescript < 3.6 don't understand `get` and `set` in `.d.ts` files](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#class-field-mitigations):

> TypeScript 3.7 emits get/set accessors in .d.ts files which can cause breaking changes for consumers on older versions of TypeScript like 3.5 and prior. TypeScript 3.6 users will not be impacted, since that version was future-proofed for this feature.

In other words, because Jlab core is now on TS 3.7 it has `get` and `set` in its `.d.ts` files. Therefore, any extension that imports Jlab packages needs to use at least TS 3.6 or it won't compile.

The error message is really cryptic, but apparently, [`ambient context` really just means `.d.ts` files](https://stackoverflow.com/a/26955889/425458).
